sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs

```diff
@@ -49,6 +44,29 @@ internal partial class TelemetryItem
             }
         }
 
+        public TelemetryItem(TelemetryItem telemetryItem, ActivitySpanId activitySpanId, ActivityKind kind, DateTimeOffset activityEventTimeStamp) :
+                        this("Exception", FormatUtcTimestamp(activityEventTimeStamp.DateTime))
+        {
+            Tags[ContextTagKeys.AiOperationParentId.ToString()] = activitySpanId.ToHexString();
+            Tags[ContextTagKeys.AiOperationId.ToString()] = telemetryItem.Tags[ContextTagKeys.AiOperationId.ToString()];
+
+            // todo: update swagger to include this key.
+            Tags["ai.user.userAgent"] = telemetryItem.Tags["ai.user.userAgent"];
+
+            // we only have mapping for server spans
+            // todo: non-server spans
+            if (kind == ActivityKind.Server)
+            {
+                Tags[ContextTagKeys.AiOperationName.ToString()] = telemetryItem.Tags[ContextTagKeys.AiOperationName.ToString()];
+                Tags[ContextTagKeys.AiLocationIp.ToString()] = telemetryItem.Tags[ContextTagKeys.AiLocationIp.ToString()];
+            }
+
+            Tags[ContextTagKeys.AiCloudRole.ToString()] = telemetryItem.Tags[ContextTagKeys.AiCloudRole.ToString()];
+            Tags[ContextTagKeys.AiCloudRoleInstance.ToString()] = telemetryItem.Tags[ContextTagKeys.AiCloudRoleInstance.ToString()];
+            Tags[ContextTagKeys.AiInternalSdkVersion.ToString()] = SdkVersionUtils.s_sdkVersion;
+            InstrumentationKey = telemetryItem.InstrumentationKey;
+        }
```

Comment:
That is a lot of `ToString()`s.
(i checked to confirm that the ToString is overriden).
https://github.com/Azure/azure-sdk-for-net/blob/1935fc360f24c6df06c0791adea6f1f6cc59b9bd/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Generated/Models/ContextTagKeys.cs#L127

I'm not sure what the `ContextTagKeys` type was intended for, but here looks like we're only using the string value.
https://github.com/Azure/azure-sdk-for-net/blob/1935fc360f24c6df06c0791adea6f1f6cc59b9bd/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Generated/Models/ContextTagKeys.cs#L20-L23

I think this is out of scope for your PR, but I think string constants would be better here.

Reply:
A similar discussion came when we first implemented it. As it was a part of AutoGenerated class with override on a string hence no further action was taken.